### PR TITLE
feat(api-markdown-documenter): Add (alpha) HTML rendering APIs

### PR DESCRIPTION
**Pull request by Josmithr**
## Summary

Adds capabilities for rendering API docs as HTML in addition to Markdown.

Updates existing Markdown rendering to fall back to this separate infra, rather than bundling its own HTML rendering (for tables, etc.).

### Breaking Changes

This PR includes a few of API breaking changes:

* Removes `RenderDocumentationNodeAsMarkdown` type alias, opting to inline the callback signature into `MarkdownRenderers`.
* Removes the `insideHtml` property from `MarkdownRenderContext`, as it is no longer needed.

## Notes for reviewers

The size of the changes here is much smaller than it appears. The bulk of the file changes are...

1. Moving existing Markdown end-to-end test snapshot assets into a new sub-directory
2. Adding new end-to-end test snapshot assets for the HTML renderer

The primary changes to look at are:

1. The new code under `renderers/html-renderer`
2. The updated code under `renderers/markdown-renderer` (which now falls back on `html-renderer`, rather than bundling its own HTML rendering logic for use in table contexts)

### CODE REVIEW

**Comment by Josmithr** [comment]
I have extracted adding `good-fences` to the package into its own PR: #17313

**Comment by Josmithr** [comment]
I have extracted the test-asset build changes into their own PR as well: #17314

File: tools/api-markdown-documenter/src/documentation-domain/DocumentNode.ts [hunk 1]
```diff
@@ -27,6 +27,11 @@ export interface DocumentNodeProps {
 
 	/**
 	 * Path to which the resulting document should be saved.
+	 *
+	 * @remarks Does not include the file extension, as this domain has no concept of what kind of file will be produced.
+	 *
+	 * @privateRemarks TODO: Maybe consider renaming this to something like "documentPath", since "filePath" would
```
Comment (Josmithr) [inline tools/api-markdown-documenter/src/documentation-domain/DocumentNode.ts:33]: This change has been made in a separate PR: #17316

File: tools/api-markdown-documenter/src/renderers/html-renderer/configuration/RenderOptions.ts [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+	renderTableRow,
+	renderUnorderedList,
+} from "../default-renderers";
+
+/**
+ * Configuration for rendering {@link DocumentationNode}s to `HTML`, specified by {@link DocumentationNode."type"}.
+ *
+ * @remarks
+ *
+ * Note: in order to be complete, this *must* include an entry for all types enumerated under {@link DocumentationNodeType}.
```
Comment (alexvy86) [inline tools/api-markdown-documenter/src/renderers/html-renderer/configuration/RenderOptions.ts:49]: Can we enforce that with typing of the keys for `Renderers`?

**Review by alexvy86 — APPROVED** [review]
Disclaimer: I didn't go through all of it. I did check pretty much all of the html-renderer files and they look pretty good.

File: tools/api-markdown-documenter/src/renderers/html-renderer/configuration/RenderOptions.ts (hunk 2 above)
Comment (Josmithr) [reply tools/api-markdown-documenter/src/renderers/html-renderer/configuration/RenderOptions.ts:49]: There is probably a way, but I'm not sure it's worth it. The "DocumentationNode" domain is extensible, so we don't technically know what the complete set is. We know what the complete set of system-provided node kinds is, but we always inject our defaults unless overridden anyways. I might just update this comment to note that if we encounter a node kind without a renderer, we will throw an error.

**Pull request merged**